pkg/app/topaz/runtime_resolver.go

```diff
@@ -33,7 +34,8 @@ func NewRuntimeResolver(
 	cfg *config.Config,
 	ctrlf *controller.Factory,
 	decisionLogger decisionlog.DecisionLogger,
-	directoryResolver resolvers.DirectoryResolver) (resolvers.RuntimeResolver, func(), error) {
+	directoryResolver resolvers.DirectoryResolver,
+	topazApp *app.Topaz) (resolvers.RuntimeResolver, func(), error) {
```

Comment:
If we change SetServiceStatus to be a func instead of a method, we do not need any changes to the runtime resolver.